docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2019-abelbriggs.md

```diff
@@ -0,0 +1,256 @@
+(if there was one), logs, etc.
+
+When reporting bugs in Mesa's drivers, it is most convenient for the Mesa developers to have a
+`.shader_test` file that they can easily plug into Piglit as a test case. Additionally, the OpenGL ES
+GraphicsFuzz worker is rather unwieldy to compile/use for desktop platforms, and so it was decided
+that the development of a Piglit/`shader_runner` worker was worth pursuing, along with a script to
+systematically convert GraphicsFuzz shader jobs into Piglit `.shader_test` files.
+
+The following PRs involve the Piglit converter script or worker.
+
+[#577](https://github.com/google/graphicsfuzz/pull/577):
+Add script to convert a GraphicsFuzz shader job to a piglit shader test
+
+[#584](https://github.com/google/graphicsfuzz/pull/584):
+Fix piglit converter script not inserting GL ES version
+
+[#587](https://github.com/google/graphicsfuzz/pull/587):
+Refactor common python script functions into one common lib
+
+[#613](https://github.com/google/graphicsfuzz/pull/613):
+Fix piglit converter not accepting blank JSON
+
+[#617](https://github.com/google/graphicsfuzz/pull/617):
+Add piglit shader_runner_gles3 worker script
+
+[#631](https://github.com/google/graphicsfuzz/pull/631):
+docs: Add documentation for piglit worker
+
+[#632](https://github.com/google/graphicsfuzz/pull/632):
+Piglit worker improvements
+
+[#638](https://github.com/google/graphicsfuzz/pull/638):
+graphicsfuzz-piglit-converter: support more uniform types
+
+In addition, a contribution was made to Piglit upstream to allow `shader_runner` to handle a case
+when unused uniform data was optimized out of a compiled shader.
+
+[shader_runner: Add command line option to ignore uniform if it isn't found in shader](https://gitlab.freedesktop.org/mesa/piglit/merge_requests/92)
+
+#### Leftovers
+
+The following issues are nits or minor problems related to the Piglit worker or converter script
+that were left unfixed due to time or knowledge constraints.
+
+[#597](https://github.com/google/graphicsfuzz/issues/597):
+Support compute shaders in graphicsfuzz_piglit_converter
+
+### Add new shaders to GraphicsFuzz's test set
+
+With the exception of compute shaders, GraphicsFuzz had a fairly limited set of five simple test
+shaders for mutation. These shaders sufficed for finding simple bugs, but were limited to GLES 1.00
+features and were unable to test the full extent of the GLSL language even with significant additions
+to the shader generator.
+
+The following PRs involve adding new shaders to GraphicsFuzz's test suite.
+
+[#605](https://github.com/google/graphicsfuzz/pull/605):
+Add new versions of GraphicsFuzz shaders that use Integer Functions
+
+[#647](https://github.com/google/graphicsfuzz/pull/647):
+New 310es shader: householder_lattice
+
+[#670](https://github.com/google/graphicsfuzz/pull/670):
+Fix making array accesses inbounds with uint index accesses
+
+### Apply GraphicsFuzz to the nouveau open-source graphics driver
```

Comment:
When you add this, it would be nice to show the bad images for the bugs, and bits of stack trace for crashes.

Reply:
See above regarding adding images.